libs/domain/cart/coupon/coupon.component.ts

```diff
@@ -0,0 +1,133 @@
+import { AuthService } from '@spryker-oryx/auth';
+import { CartComponentMixin, CartService } from '@spryker-oryx/cart';
+import { resolve } from '@spryker-oryx/di';
+import { ContentMixin } from '@spryker-oryx/experience';
+import { LocaleService } from '@spryker-oryx/i18n';
+import { NotificationService } from '@spryker-oryx/site';
+import { AlertType } from '@spryker-oryx/ui';
+import { ButtonSize, ButtonType } from '@spryker-oryx/ui/button';
+import { IconTypes } from '@spryker-oryx/ui/icon';
+import { ColorType } from '@spryker-oryx/ui/link';
+import { ClearIconAppearance } from '@spryker-oryx/ui/searchbox';
+import { hydrate, signal } from '@spryker-oryx/utilities';
+import { LitElement, TemplateResult, html } from 'lit';
+import { query, state } from 'lit/decorators.js';
+import { repeat } from 'lit/directives/repeat.js';
+import { couponStyles } from './coupon.styles';
+
+@hydrate({ event: ['window:load'] })
+export class CouponComponent extends CartComponentMixin(
+  ContentMixin(LitElement)
+) {
+  static styles = couponStyles;
+
+  @state() hasError = false;
+  @state() errorMessage: '' | string = '';
+
+  @query('input[name=coupon]') coupon?: HTMLInputElement;
+
+  protected authService = resolve(AuthService);
+  protected cartService = resolve(CartService);
+  protected notificationService = resolve(NotificationService);
+  protected localeService = resolve(LocaleService);
+
+  protected isAuthenticated = signal(this.authService.isAuthenticated());
+
+  private successMessage: '' | string = '';
+
+  protected override render(): TemplateResult | void {
+    if (this.$isEmpty() || !this.isAuthenticated()) {
```

Comment:
We seem to have guest cart integration for coupons, see https://docs.spryker.com/docs/pbc/all/discount-management/202307.0/base-shop/manage-using-glue-api/glue-api-manage-discount-vouchers-in-guest-carts.html. I'd hope we can leverage this at adapter or service level.